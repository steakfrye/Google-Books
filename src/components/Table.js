import React, { Component } from 'react';
import BookFeed from './BookFeed';
import ModalBox from './ModalBox';

export class Table extends Component {
  constructor(props) {
    super(props);

    this.state = {
      book: {},
      showComponent: false
    }
    this.showModal = this.showModal.bind(this);
  }

  // Component passes down to <BookItem />
  showModal(book) {
    this.setState({ book: book, showComponent: true })
  }

  render() {
    return (
      <div className="container">
        {this.state.showComponent ?
        <ModalBox book={this.state.book} /> :
        null}
        <h3>Click on a book for more info.</h3>
        <table className="table">
          <thead>
            <tr>
              <th scope="col"></th>
              <th scope="col">Title</th>
              <th scope="col">Author</th>
              <th scope="col">Publisher</th>
            </tr>
          </thead>
          <tbody>
            <BookFeed books={this.props.books} showModal={this.showModal}/>
          </tbody>
        </table>
      </div>
    );
  }
}

export default Table;
