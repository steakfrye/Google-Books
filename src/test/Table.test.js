import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Table from '../components/Table';
import books from './mockData';

configure({ adapter: new Adapter() });

describe('<Table />', () => {

  it('renders book data', () => {
    const tree = renderer
      .create(shallow(<Table books={books} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
