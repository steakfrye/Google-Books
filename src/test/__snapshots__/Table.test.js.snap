// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders book data 1`] = `
<div
  className="container"
>
  <h3>
    Click on a book for more info.
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        />
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Author
        </th>
        <th
          scope="col"
        >
          Publisher
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="toggleInput"
        data-target="#exampleModal"
        data-toggle="modal"
        onClick={[Function]}
      >
        <td>
          <img
            alt="Collaborative Quilting"
            src="http://tinyurl.com/55"
          />
        </td>
        <td>
          Collaborative Quilting
        </td>
        <td>
          Freddy Moran, Gwen Marston
        </td>
        <td>
          Sterling Publishing Company, Inc.
        </td>
      </tr>
      <tr
        className="toggleInput"
        data-target="#exampleModal"
        data-toggle="modal"
        onClick={[Function]}
      >
        <td>
          <img
            alt="Collaborative Pizza"
            src="http://tinyurl.com/54"
          />
        </td>
        <td>
          Collaborative Pizza
        </td>
        <td>
          James Morgan, Greg Mangus
        </td>
        <td>
          The Publishing Company, Inc.
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Show more...
    </button>
  </div>
</div>
`;
